ccl/changefeedccl: TestDistSenderRangeFeedPopulatesVirtualTable failed

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/d2010221-d834-46a0-8d93-28693a66cbb7?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [fdb56245f7d6367aa1db9adccf80540809998f74](https://github.com/cockroachdb/cockroach/commits/fdb56245f7d6367aa1db9adccf80540809998f74):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4253384554
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9621: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4253384554
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.28s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage
    helpers_test.go:1389: making server as secondary tenant
    helpers_test.go:1468: making cloudstorage feed factory
    helpers_test.go:1196: did not force enriched envelope for CREATE CHANGEFEED FOR table_a; because *changefeedccl.cloudFeedFactory is not supported
    testfeed_test.go:342: creating external connection
    testfeed_test.go:345: ran create external connection
    changefeed_test.go:9609: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9609
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1262
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9606
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9605
        	            				pkg/ccl/changefeedccl/helpers_test.go:1648
        	            				pkg/ccl/changefeedccl/helpers_test.go:1684
        	Error:      	[]int{3, 5, 20, 6, 3, 46, 3, 5, 32} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage (2.27s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53614

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/8a1d447c-cee8-409e-a173-63003e8ec1fb?testReportRun=5&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [71da658c99ed1f646a1078b0a5046e4b6e5f8c04](https://github.com/cockroachdb/cockroach/commits/71da658c99ed1f646a1078b0a5046e4b6e5f8c04):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4120725771
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9646: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4120725771
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.34s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/kafka
    helpers_test.go:1390: making server as secondary tenant
    helpers_test.go:1469: making kafka feed factory
    changefeed_test.go:9634: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9634
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1263
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9631
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9630
        	            				pkg/ccl/changefeedccl/helpers_test.go:1649
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	[]int{3, 5, 20, 6, 3, 46, 3, 5, 32} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/kafka
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/kafka (2.33s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/e0a1f0d3-bb20-4942-8d64-0a1f0e6e4353?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [82759997da526f08ead6de93e47fb30819edf10e](https://github.com/cockroachdb/cockroach/commits/82759997da526f08ead6de93e47fb30819edf10e):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable3090710876
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9646: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable3090710876
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.85s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/webhook
    helpers_test.go:1390: making server as secondary tenant
    helpers_test.go:1469: making webhook feed factory
    changefeed_test.go:9634: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9634
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1263
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9631
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9630
        	            				pkg/ccl/changefeedccl/helpers_test.go:1649
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	[]int{5, 20, 6, 3, 46, 3, 5, 32, 3} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/webhook
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/webhook (2.83s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/7ab32420-06ec-4aa9-915d-6565ac079a94?testReportRun=4&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [f8a09a5c1dbcbfcb537ba4050c6061ef6d917687](https://github.com/cockroachdb/cockroach/commits/f8a09a5c1dbcbfcb537ba4050c6061ef6d917687):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable1005450869
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9646: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable1005450869
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.74s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/kafka
    helpers_test.go:1390: making server as secondary tenant
    helpers_test.go:1469: making kafka feed factory
    changefeed_test.go:9634: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9634
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1263
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9631
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9630
        	            				pkg/ccl/changefeedccl/helpers_test.go:1649
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	[]int{32, 3, 5, 20, 6, 3, 46, 5, 3} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/kafka
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/kafka (2.72s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/97759b8c-6ccc-4eeb-9b51-53bdf97bdaeb?testReportRun=10&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [f8a09a5c1dbcbfcb537ba4050c6061ef6d917687](https://github.com/cockroachdb/cockroach/commits/f8a09a5c1dbcbfcb537ba4050c6061ef6d917687):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable1625319612
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9646: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable1625319612
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.50s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage
    helpers_test.go:1390: making server as secondary tenant
    helpers_test.go:1469: making cloudstorage feed factory
    changefeed_test.go:9634: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9634
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1263
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9631
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9630
        	            				pkg/ccl/changefeedccl/helpers_test.go:1649
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	[]int{6, 46, 5, 3, 32, 3, 5, 20, 3} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/cloudstorage (2.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestDistSenderRangeFeedPopulatesVirtualTable [failed](https://mesolite.cluster.engflow.com/invocations/default/e7c75737-b2fe-455b-82ed-a9b18148fc7a?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [3d5ef9464d282f18aaab842d2dc0bd5d49353f41](https://github.com/cockroachdb/cockroach/commits/3d5ef9464d282f18aaab842d2dc0bd5d49353f41):


```
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4216375569
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9646: -- test log scope end --
test logs left over in: outputs.zip/logTestDistSenderRangeFeedPopulatesVirtualTable4216375569
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable (2.43s)
=== RUN   TestDistSenderRangeFeedPopulatesVirtualTable/kafka
    helpers_test.go:1390: making server as secondary tenant
    helpers_test.go:1469: making kafka feed factory
    changefeed_test.go:9634: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9634
        	            				pkg/ccl/changefeedccl/testfeed_test.go:876
        	            				pkg/ccl/changefeedccl/helpers_test.go:1263
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9631
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:346
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9630
        	            				pkg/ccl/changefeedccl/helpers_test.go:1649
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	[]int{3, 5, 32, 3, 5, 20, 6, 3, 46} does not contain 108
        	Test:       	TestDistSenderRangeFeedPopulatesVirtualTable/kafka
        	Messages:   	user adminUser should see table 108
--- FAIL: TestDistSenderRangeFeedPopulatesVirtualTable/kafka (2.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedPopulatesVirtualTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -9583,8 +9583,6 @@ func TestDistSenderRangeFeedPopulatesVirtualTable(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 152060)
-
 	scanner := keysutil.MakePrettyScanner(nil, nil)
 
 	observeTables := func(sqlDB *sqlutils.SQLRunner, codec keys.SQLCodec) []int {
@@ -9627,6 +9625,10 @@ func TestDistSenderRangeFeedPopulatesVirtualTable(t *testing.T) {
 			cf = feed(t, f, `CREATE CHANGEFEED FOR table_a;`)
 		})
 		defer closeFeed(t, cf)
+
+		// We need to ensure that the reason the user doesn't see the table
+		// is not because the rangefeed hasn't started yet.
+		waitForHighwater(t, cf.(cdctest.EnterpriseTestFeed), s.Server.JobRegistry().(*jobs.Registry))
 
 		for _, c := range cases {
 			testutils.SucceedsSoon(t, func() error {

